python/private/common/py_runtime_rule.bzl

```diff
@@ -34,12 +34,33 @@ def _py_runtime_impl(ctx):
         for t in ctx.attr.files
     ])
 
+    runfiles = ctx.runfiles()
+
     hermetic = bool(interpreter)
     if not hermetic:
         if runtime_files:
             fail("if 'interpreter_path' is given then 'files' must be empty")
         if not paths.is_absolute(interpreter_path):
             fail("interpreter_path must be an absolute path")
+    else:
+        interpreter_di = ctx.attr.interpreter[DefaultInfo]
+
+        if _is_singleton_depset(interpreter_di.files):
+            interpreter = interpreter_di.files.to_list()[0]
+        elif interpreter_di.files_to_run and interpreter_di.files_to_run.executable:
+            interpreter = interpreter_di.files_to_run.executable
+            runfiles.merge(interpreter_di.default_runfiles)
+
+            runtime_files = depset(
+                direct = [interpreter_di.files_to_run.runfiles_manifest, interpreter_di.files_to_run.repo_mapping_manifest],
```

Comment:
files_to_run.runfiles_manifest can be None if certain flags are set, so this needs a guard, at the least.

However, I don't think either the manifest or repo_mapping need to be manually propagated; you can remove that part.